### PR TITLE
Find_value working

### PR DESCRIPTION
**Pull request by Hampusholmstrom**

### CODE REVIEW

**Comment by Janaza** [comment]
sweet

**Comment by Hampusholmstrom** [comment]
Find value primitive now returns data of given hash. 
Find value iterative returns the value if key value pair is available somewhere in the network.

Added STORE to cli for testing.
Added CONTACTS to cli to show k closest contacts of node.

File: kademlia.go [hunk 1]
```diff
@@ -93,22 +93,164 @@ func (kademlia *Kademlia) LookupContact(network Network, result chan []Contact,
...
-	go kademlia.LookupContact(*network, ch, contact)
-	done :=  <- ch
-	for _, c := range done{
-		kademlia.net.SendFindDataMessage(hash, &c)
+
+func (kademlia *Kademlia) LookupData(network Network, target Contact, hash string) string {
+	alpha := 3
+	value := make(chan string)
+	found := make(chan []Contact)
+	//var closestNode Contact
```
Comment (Janaza) [inline kademlia.go]: rm comments

File: kademlia.go [hunk 2]
```diff
@@ -93,22 +93,164 @@ func (kademlia *Kademlia) LookupContact(network Network, result chan []Contact,
...
+			runningRoutines = 0
+			return y
+		}
+		runningRoutines++
+		/*
+		for _, i := range x{
+			fmt.Println(i.ID)
+		}
+
+		 */
```
Comment (Janaza) [inline kademlia.go]: ^

File: kademlia.go [hunk 3]
```diff
@@ -93,22 +93,164 @@ func (kademlia *Kademlia) LookupContact(network Network, result chan []Contact,
...
+			}
+			runningRoutines = 0
+			return y
+		}
+		/*
+		for _, i := range x{
+			fmt.Println(i.ID)
+		}
+
+		 */
```
Comment (Janaza) [inline kademlia.go]: ^

Comment (Janaza) [reply kademlia.go]: osv.

**Pull request merged**